runtime/autoload/man.vim

```diff
@@ -54,44 +54,43 @@ function! man#open_page(count, count1, mods, ...) abort
   call s:push_tag()
   let bufname = 'man://'.name.(empty(sect)?'':'('.sect.')')
   if a:mods !~# 'tab' && s:find_man()
-    if s:manwidth() ==# getbufvar(bufname, 'manwidth')
-      silent execute 'buf' bufname
-      call man#set_window_local_options()
-      keepjumps 1
-      return
-    endif
     noautocmd execute 'edit' bufname
-  elseif s:manwidth() ==# getbufvar(bufname, 'manwidth')
-    execute a:mods 'split' bufname
-    call man#set_window_local_options()
-    keepjumps 1
-    return
   else
     noautocmd execute a:mods 'split' bufname
   endif
+  let b:man_sect = sect
   call s:read_page(path)
 endfunction
 
 function! man#read_page(ref) abort
   try
-    let [sect, name] = s:extract_sect_and_name_ref(a:ref)
-    " The third element is the path.
-    call s:read_page(s:verify_exists(sect, name)[2])
+    let [sect, name] = man#extract_sect_and_name_ref(a:ref)
+    let [b:man_sect, name, path] = s:verify_exists(sect, name)
   catch
-    call s:error(v:exception)
+    " call to s:error() is unnecessary
     return
   endtry
+  call s:read_page(path)
 endfunction
 
 function! s:read_page(path) abort
   setlocal modifiable
   setlocal noreadonly
   keepjumps %delete _
-  let b:manwidth = s:manwidth()
   " Ensure Vim is not recursively invoked (man-db does this)
   " by forcing man to use cat as the pager.
   " More info here http://comments.gmane.org/gmane.editors.vim.devel/29085
-  silent execute 'read !env MANPAGER=cat MANWIDTH='.b:manwidth s:man_cmd a:path
+  let cmd = 'read !env MANPAGER=cat'
```

Comment:
Just how it always was. Wouldn't the redraw occur with `system()` anyway because you have to set the contents of the buffer to whatever `system()` outputs?